src/scripts/consume_logs/consume_logs.py

```diff
@@ -0,0 +1,312 @@
+    )
+    parser.add_argument(
+            "--end-datetime",
+            help=(
+                "Default is \"now\". Query end time (local time) expressed "
+                "in a format parseable by datetime.datetime.strptime. This "
+                "argument should be formatted as `--time-format`."),
+    )
+    parser.add_argument(
+            "--datetime-format",
+            help="The time format to use with datetime.datetime.strptime",
+            default="%Y-%m-%d %H:%M:%S",
+    )
+    args = parser.parse_args()
+    return args
+
+def get_seconds_since_epoch(datetime_str: str, datetime_format: str):
+    """
+    Returns the seconds since epoch for a specific datetime (local time)
+
+    Args:
+        datetime_str (str): A datetime represented by a format parseable by
+            datetime.datetime.strptime.
+        datetime_format (str): The datetime.datetime.strptime format used
+            to parse `datetime`.
+
+    Returns:
+        (int) The number of seconds since January 1, 1970 (local time)
+    """
+    if datetime_str is None:
+        parsed_datetime = datetime.datetime.now()
+    else:
+        parsed_datetime = datetime.datetime.strptime(datetime_str, datetime_format)
+    seconds_since_epoch = int(parsed_datetime.timestamp())
+    return seconds_since_epoch
+
+def query_logs(
+        log_group_name: str,
+        query_string: str,
+        start_unix_time: int,
+        end_unix_time: int,
+        **kwargs: dict,
+) -> list:
+    """
+    Query a CloudWatch log group.
+
+    Args:
+        log_group_name (str): The log group on which to perform the query.
+        query_string (str): The query string to use.
+        start_unix_time (int): The beginning of the time range to query. The range
+            is inclusive, so the specified start time is included in the query.
+            Specified as epoch (Unix) time.
+        end_unix_time (int): The end of the time range to query. The range
+            is inclusive, so the specified end time is included in the query.
+            Specified as epoch (Unix) time.
+        kwargs (dict): Used during tests to include boto client stubs
+
+    Returns:
+        (list) The results of the query. Empty if no matches found.
+
+    Raises:
+        UserWarning: If the call to the get query results API does not succeed.
+    """
+    if "logs_client" in kwargs:
+        logs_client = kwargs["logs_client"]
+    else:
+        logs_client = boto3.client("logs")
+    start_query_response = logs_client.start_query(
+            logGroupName=log_group_name,
+            startTime=start_unix_time,
+            endTime=end_unix_time,
+            queryString=query_string
+    )
+    query_response = logs_client.get_query_results(
+            queryId=start_query_response["queryId"]
+    )
+    _check_for_failed_query(query_response)
+    while query_response["status"] != "Complete":
+        time.sleep(1)
+        query_response = logs_client.get_query_results(
```

Comment:
Nit: There's probably some fun stuff you can do with https://docs.python.org/3/library/asyncio.html here. But for a PoC, that may be overkill